src/names/applications.cpp

```diff
@@ -183,4 +185,35 @@ IsValidJSONOrEmptyString (const std::string& text){
     UniValue v;
 
     return text.empty() || v.read(text);
+}    
+
+bool
+IsMinimalJSONOrEmptyString (const std::string& text){
+    UniValue v;
+    if(text.empty()){
+        return true;
+    } 
+
+    if(!v.read(text)){ 
+        return false;
+    } 
+
+    std::string minimalJSON = GetMinimalJSON(text);
+    
+    bool matches = (text == minimalJSON);
```

Comment:
Both of `minimalJSON` and `matches` can be declared as `const`, and I think it is good style to do so in this case.  Also I would rename `matches` maybe to `isMinimal` to make the code more readable.